lib/Onnxifi/GlowOnnxManager.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "GlowOnnxManager.h"
+
+#include "Base.h"
+
+#include <mutex>
+
+namespace glow {
+namespace onnxifi {
+GlowOnnxManager &GlowOnnxManager::get() {
+  static GlowOnnxManager manager;
+  return manager;
+}
+
+void GlowOnnxManager::addBackendId(BackendIdPtr backendId) {
+  assert(!isValid(backendId) && backendId != nullptr);
```

Comment:
`isValid` already does this but in this case, we want to make sure that the reason why backendId is invalid is not because it's null but because it's actually not been added already so the extra check is needed (or this could be written without `isValid` which would probably be better actually)